javascript-typing-tutor/main.js

```diff
@@ -0,0 +1,49 @@
+var charNodes = document.querySelectorAll('span');
+var currentIndex = 0;
+document.addEventListener('keydown', typingTest);
+
+function typingTest(event) {
+  if (currentIndex === charNodes.length) {
+    alert('DONE! Restarting Test....');
+    window.location.reload();
```

Comment:
You should **_never_** use `location.reload()`, it reloads the page entirely, which is one of the most inefficient actions you can take. Instead, you should reset each variable value and reset the classes on all elements to bring the page back to how it was when the page first loaded